hiclass/Explainer.py

```diff
@@ -0,0 +1,283 @@
+"""Explainer API for explaining predictions using shapley values."""
+
+from copy import deepcopy
+from joblib import Parallel, delayed
+import numpy as np
+from sklearn.utils.validation import check_array, check_is_fitted
+from hiclass import (
+    LocalClassifierPerParentNode,
+    LocalClassifierPerNode,
+    LocalClassifierPerLevel,
+    HierarchicalClassifier,
+)
+
+try:
+    import xarray as xr
+except ImportError:
+    xarray_installed = False
+else:
+    xarray_installed = True
+
+try:
+    import shap
+except ImportError:
+    shap_installed = False
+else:
+    shap_installed = True
+
+
+class Explainer:
+    """Explainer class for returning shap values for each of the three hierarchical classifiers."""
+
+    def __init__(
+        self,
+        hierarchical_model: HierarchicalClassifier.HierarchicalClassifier,
+        data: None,
+        n_jobs: int = 1,
+        algorithm: str = "auto",
+        mode: str = "",
+    ):
+        """
+        Initialize the SHAP explainer for a hierarchical model.
+
+        Parameters
+        ----------
+        hierarchical_model : HierarchicalClassifier
+            The hierarchical classification model to explain.
+        data : array-like or None, default=None
+            The dataset used for creating the SHAP explainer.
+        n_jobs : int, default=1
+            The number of jobs to run in parallel.
+        algorithm : str, default="auto"
+            The algorithm to use for SHAP explainer. Possible values are 'linear', 'tree', 'auto', 'permutation'. or 'partition'
+        mode : str, default=""
+            The mode of the SHAP explainer. Can be 'tree', 'gradient', 'deep', 'linear', or '' for default SHAP explainer.
+
+        Examples
+        --------
+        >>> from sklearn.ensemble import RandomForestClassifier
+        >>> import numpy as np
+        >>> from hiclass import LocalClassifierPerParentNode, Explainer
+        >>> rfc = RandomForestClassifier()
+        >>> lcppn = LocalClassifierPerParentNode(local_classifier=rfc, replace_classifiers=False)
+        >>> x_train = np.array([[1, 3], [2, 5]])
+        >>> y_train = np.array([[1, 2], [3, 4]])
+        >>> x_test = np.array([[4, 6]])
+        >>> lcppn.fit(x_train, y_train)
+        >>> explainer = Explainer(lcppn, data=x_train, mode="tree")
+        >>> explanations = explainer.explain(x_test)
+        <xarray.Dataset>
+        Dimensions:          (class: 3, sample: 1, level: 2, feature: 2)
+        Coordinates:
+          * class            (class) <U1 '1' '3' '4'
+          * level            (level) int64 0 1
+        Dimensions without coordinates: sample, feature
+        Data variables:
+            node             (sample, level) <U13 'hiclass::root' '3'
+            predicted_class  (sample, level) <U24 '3' '3::HiClass::Separator::4'
```

Comment:
```suggestion
            predicted_class  (sample, level) <U24 '3' '4'
```
is it possible to remove the separator from the output shown to the user? It is meant to be for internal use only and transparent to the user. Besides, I am planning to remove it in the future.